numpy/random/src/distributions/distributions.c

```diff
@@ -696,7 +658,7 @@ static RAND_INT_TYPE random_poisson_ptrs(bitgen_t *bitgen_state, double lam) {
     /* log(V) == log(0.0) ok here */
     /* if U==0.0 so that us==0.0, log is ok since always returns */
     if ((log(V) + log(invalpha) - log(a / (us * us) + b)) <=
-        (-lam + k * loglam - loggam(k + 1))) {
+        (-lam + k * loglam - logfactorial((int64_t)k))) {
```

Comment:
k is already int64_t from `(RAND_INT_TYPE)` above.

Reply:
But if `NP_RANDOM_LEGACY` is defined, `RAND_INT_TYPE` will be a C long, which could be 32 bits.

Reply:
But in that case, C will implicitly cast the argument to `int64_t`, and that is safe, so perhaps that explicit cast is unnecessarily pedantic.  I can remove it if you prefer.

Reply:
If this function is every used in mtrand then it cannot change.  So either RAND_INT_TYPE is always int64_t, or a copy is needed of this function in legacy-distributions.

Reply:
`random_poisson` is used in legacy-distirbutions, and so this function cannot change.  

All of poisson will need to be copied or legacy_random_poisson in legacy-distributions to guarantee exact byte stream compatibility AFAICT.

Reply:
Ah, right.  I don't think making the copy of poisson is worth it just for the change from `loggam` to `logfactorial`.  I'll undo that change.